openreview/venue/venue.py

```diff
@@ -1009,6 +1015,125 @@ def set_SAC_ethics_review_process(self, sac_ethics_flag_duedate=None):
     def open_reviewer_recommendation_stage(self, start_date=None, due_date=None, total_recommendations=7):
         self.invitation_builder.set_reviewer_recommendation_invitation(start_date, due_date, total_recommendations)
 
+    def ithenticate_create_and_upload_submission(self):
+        if not self.iThenticatePlagiarismCheck:
+            raise openreview.OpenReviewException('iThenticatePlagiarismCheck is not enabled for this venue.')
+        
+        self.invitation_builder.set_iThenticate_plagiarism_check_invitation()
+
+        iThenticate_client = openreview.api.iThenticateClient(self.iThenticatePlagiarismCheckApiKey, self.iThenticatePlagiarismCheckApiBaseUrl)
+        
+        submissions = self.get_submissions()
+        for submission in submissions:
+            # Create a new submission ID in iThenticate
+            # TODO - Decide what should go in metadata.group_context.owners
+            res = iThenticate_client.create_submission(
+                owner=submission.content["authorids"]["value"][0],
+                title=submission.content["title"]["value"],
+                timestamp=datetime.datetime.fromtimestamp(submission.tcdate/1000, tz=datetime.timezone.utc).strftime('%Y-%m-%dT%H:%M:%SZ'),
+                owner_first_name=submission.content["authors"]["value"][0].split(
+                    " ", 1
+                )[0],
+                owner_last_name=submission.content["authors"]["value"][0].split(
```

Comment:
you can use the pretty_id of the signature:

```
name = openreview.tools.pretty_id(submission.signatures[0])
first = name.split(' ')[0]
last = name.split(' ')[1:]
```

Reply:
I looked at some pre-existing notes and the signatures field doesn't have the user ID/name of the author it only contains a group ID for the authors.